src/controller/java/AndroidCallbacks.cpp

```diff
@@ -406,25 +413,39 @@ void ReportCallback::OnEventData(const app::EventHeader & aEventHeader, TLV::TLV
 #endif
 
     // Create TLV byte array to pass to Java layer
-    size_t bufferLen                  = readerForJavaTLV.GetRemainingLength() + readerForJavaTLV.GetLengthRead();
+    TLV::TLVReader readerForJavaTLV;
+    readerForJavaTLV.Init(*apData);
+    size_t bufferLen                  = UINT32_MAX;
```

Comment:
Allocating a 4GB buffer does not seem right ....

Reply:
fixed, thanks for catching up it.